src/content/learn/reacting-to-input-with-state.md

```diff
@@ -1,37 +1,38 @@
 ---
-title: Reacting to Input with State
+title: Merespon Masukan dengan State
 ---
 
 <Intro>
 
-React provides a declarative way to manipulate the UI. Instead of manipulating individual pieces of the UI directly, you describe the different states that your component can be in, and switch between them in response to the user input. This is similar to how designers think about the UI.
+React menyediakan cara deklaratif untuk memanipulasi UI. Dibandingan dengan memanipulasi bagian-bagian UI secara langsung, Anda dapat membuat berbagai *state* komponen, dan menggunakan *state* tersebut sebagai respons terhadap masukan pengguna. Cara ini mirip dengan bagaimana desainer memikirkan tentang UI.
 
 </Intro>
 
 <YouWillLearn>
 
-* How declarative UI programming differs from imperative UI programming
-* How to enumerate the different visual states your component can be in
-* How to trigger the changes between the different visual states from code
+* Bagaimana pemrograman UI deklaratif berbeda dari pemrograman UI imperatif
+* Bagaimana cara menjabarkan berbagai *state* visual yang berbeda pada komponen Anda
+* Bagaimana cara memicu perubahan ke berbagai *state*
 
 </YouWillLearn>
 
-## How declarative UI compares to imperative {/*how-declarative-ui-compares-to-imperative*/}
+## Membandingkan UI deklaratif dengan imperatif {/*how-declarative-ui-compares-to-imperative*/}
 
-When you design UI interactions, you probably think about how the UI *changes* in response to user actions. Consider a form that lets the user submit an answer:
+Ketika Anda mendesain interaksi UI, Anda mungkin berpikir tentang bagaimana UI *berubah* dalam menanggapi tindakan pengguna. Pertimbangkan sebuah formulir yang memungkinkan pengguna mengirimkan jawaban:
 
-* When you type something into the form, the "Submit" button **becomes enabled.**
-* When you press "Submit", both the form and the button **become disabled,** and a spinner **appears.**
-* If the network request succeeds, the form **gets hidden,** and the "Thank you" message **appears.**
-* If the network request fails, an error message **appears,** and the form **becomes enabled** again.
+* Anda mengetikan sesuatu kedalam formulir, maka tombol kirim menjadi **aktif**
```

Comment:
```suggestion
* Ketika Anda mengetikkan sesuatu kedalam formulir, maka tombol "Kirim" **menjadi aktif**
```
1. Please retain the `when` word translation.
2. `mengetikkan` is formed by `me(ng)`-`ketik`-`kan`.
3. `ke` is a [preposition](https://ejaan.kemdikbud.go.id/eyd/penulisan-kata/kata-depan/), it should be separated.
4. Retain the double quotes and capitalization on `"Submit"` to `"Kirim"`
5. Include `menjadi` inside the bold formatting, since `become` was also in bold.